### ISSUE TITLE
[Filebeat] Set filebeat.registry_flush default to 1s

### ISSUE THREAD

**Issue by andrewkroh** [elastic/beats#30279]
**Describe the enhancement:**

Change the default `filebeat.registry_flush` be a positive value such that some amount of waiting occurs between registry updates. The current value catches a lot of users by surprise with high CPU and high disk IO because every registry state update triggers a registry checkpoint operation.

For busy Filebeat instances I've seen 10-20 checkpoint operations per second. Each chechpoint requires multiple disk operations - write tmp file, rename to checkpoint file, sync checkpoint to disk, update marker file, sync marker file, delete old date files, delete tmp checkpoint file. For use-cases that employ the `include_lines`/`exclude_lines` I've seen 1400 registry writes per second (we warn users about this is the docs, but they usually miss it).

**Describe a specific use case for the enhancement or feature:**

By setting the value to 1s (or some non-zero value) Filebeat will throttle the checkpoint operations that trigger disk activity to once per second. When Filebeat stops any pending updates are flushed to disk. If Filebeat crashes or the machine loses power there would be 1s of registry states lost which could result in duplicate events on restart. I think it's much better to optimize for user experience in the normal case rather then prevent registry state loss during a crash (Filebeat should never crash).

Another reason to throttle disk writes it to limit the SSD wear.

**Comment by kvch** [comment]
I agree with the proposal, but I am afraid that once we change the default value, we will get questions about why events get duplicated. We should suggest using some form of deduplication in Filebeat by default if we change this. Maybe we could even write a documentation page about the Filebeat registry? We seem to get loads of questions and some of them could be avoided if we had a somethings that explained the internals of the registry and help users fine-tune their config.
